Fix RuntimeException thrown in generator due to illegal script

**Causing script**
```
const b = 
    true ||
    || false
;
```

**Stacktrace**
```
Severe error occurred in transpiler=es message=element to process may not be null.
java.lang.IllegalArgumentException: element to process may not be null
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1241)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.processBinaryExpression(PrettyPrinterEcmaScript.java:1404)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseBinaryLogicalExpression(PrettyPrinterEcmaScript.java:1083)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseBinaryLogicalExpression(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:1378)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1243)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.processBinaryExpression(PrettyPrinterEcmaScript.java:1400)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseBinaryLogicalExpression(PrettyPrinterEcmaScript.java:1083)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseBinaryLogicalExpression(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:1378)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1243)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseVariableDeclaration(PrettyPrinterEcmaScript.java:440)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseVariableDeclaration(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:480)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1216)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseVariableStatement(PrettyPrinterEcmaScript.java:408)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseVariableStatement(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:450)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1243)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1226)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseScript(PrettyPrinterEcmaScript.java:101)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseScript(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:96)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:58)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.append(PrettyPrinterEcmaScript.java:55)
	at org.eclipse.n4js.transpiler.print.PrettyPrinter.print(PrettyPrinter.java:60)
	at org.eclipse.n4js.transpiler.PrettyPrintingStep.print(PrettyPrintingStep.java:37)
	at org.eclipse.n4js.transpiler.AbstractTranspiler.prettyPrint(AbstractTranspiler.java:273)
	at org.eclipse.n4js.transpiler.AbstractTranspiler.transpile(AbstractTranspiler.java:169)
	at org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator.internalDoGenerate(EcmaScriptSubGenerator.java:163)
	at org.eclipse.n4js.generator.AbstractSubGenerator.doGenerate(AbstractSubGenerator.java:153)
	at org.eclipse.n4js.generator.N4JSCompositeGenerator.doGenerate(N4JSCompositeGenerator.java:52)
	at org.eclipse.xtext.generator.GeneratorDelegate.doGenerate(GeneratorDelegate.java:46)
	at org.eclipse.xtext.generator.GeneratorDelegate.generate(GeneratorDelegate.java:35)
	at org.eclipse.n4js.xtext.ide.server.build.XStatefulIncrementalBuilder.generate(XStatefulIncrementalBuilder.java:334)
	at org.eclipse.n4js.xtext.ide.server.build.XStatefulIncrementalBuilder.buildClustered(XStatefulIncrementalBuilder.java:265)
	at org.eclipse.n4js.xtext.ide.server.build.XStatefulIncrementalBuilder.lambda$1(XStatefulIncrementalBuilder.java:155)
	at org.eclipse.xtext.xbase.lib.internal.FunctionDelegate.apply(FunctionDelegate.java:43)
	at com.google.common.collect.Lists$TransformingRandomAccessList$1.transform(Lists.java:614)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:47)
	at java.base/java.util.AbstractCollection.toArray(AbstractCollection.java:146)
	at java.base/java.util.ArrayList.addAll(ArrayList.java:670)
	at org.eclipse.n4js.xtext.ide.server.build.XClusteringStorageAwareResourceLoader.executeClustered(XClusteringStorageAwareResourceLoader.java:107)
	at org.eclipse.n4js.xtext.ide.server.build.XBuildContext.executeClustered(XBuildContext.java:62)
	at org.eclipse.n4js.xtext.ide.server.build.XStatefulIncrementalBuilder.launch(XStatefulIncrementalBuilder.java:154)
	at org.eclipse.n4js.xtext.ide.server.build.XIncrementalBuilder.build(XIncrementalBuilder.java:60)
	at org.eclipse.n4js.xtext.ide.server.build.XIncrementalBuilder.build(XIncrementalBuilder.java:42)
	at org.eclipse.n4js.xtext.ide.server.build.ProjectBuilder.doBuild(ProjectBuilder.java:360)
	at org.eclipse.n4js.ide.server.build.N4JSProjectBuilder.doBuild(N4JSProjectBuilder.java:81)
	at org.eclipse.n4js.xtext.ide.server.build.ProjectBuilder.doIncrementalBuild(ProjectBuilder.java:327)
	at org.eclipse.n4js.xtext.ide.server.build.XWorkspaceBuilder.doIncrementalBuild(XWorkspaceBuilder.java:536)
	at org.eclipse.n4js.xtext.ide.server.build.XWorkspaceBuilder.doIncrementalWorkspaceUpdateAndBuild(XWorkspaceBuilder.java:456)
	at org.eclipse.n4js.xtext.ide.server.build.BuilderFrontend.lambda$6(BuilderFrontend.java:191)
	at org.eclipse.n4js.ide.server.concurrent.N4JSQueuedExecutorService.lambda$0(N4JSQueuedExecutorService.java:35)
	at org.eclipse.n4js.xtext.ide.server.QueuedExecutorService$QueuedTask.run(QueuedExecutorService.java:124)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Build ABORTED due to exception: Severe error occurred in transpiler=es message=element to process may not be null.
ERROR - error during queued task: 
org.eclipse.n4js.generator.GeneratorException: Severe error occurred in transpiler=es message=element to process may not be null.
	at org.eclipse.n4js.generator.GeneratorExceptionHandler.handleError(GeneratorExceptionHandler.java:27)
	at org.eclipse.n4js.generator.AbstractSubGenerator.doGenerate(AbstractSubGenerator.java:205)
	at org.eclipse.n4js.generator.N4JSCompositeGenerator.doGenerate(N4JSCompositeGenerator.java:52)
	at org.eclipse.xtext.generator.GeneratorDelegate.doGenerate(GeneratorDelegate.java:46)
	at org.eclipse.xtext.generator.GeneratorDelegate.generate(GeneratorDelegate.java:35)
	at org.eclipse.n4js.xtext.ide.server.build.XStatefulIncrementalBuilder.generate(XStatefulIncrementalBuilder.java:334)
	at org.eclipse.n4js.xtext.ide.server.build.XStatefulIncrementalBuilder.buildClustered(XStatefulIncrementalBuilder.java:265)
	at org.eclipse.n4js.xtext.ide.server.build.XStatefulIncrementalBuilder.lambda$1(XStatefulIncrementalBuilder.java:155)
	at org.eclipse.xtext.xbase.lib.internal.FunctionDelegate.apply(FunctionDelegate.java:43)
	at com.google.common.collect.Lists$TransformingRandomAccessList$1.transform(Lists.java:614)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:47)
	at java.base/java.util.AbstractCollection.toArray(AbstractCollection.java:146)
	at java.base/java.util.ArrayList.addAll(ArrayList.java:670)
	at org.eclipse.n4js.xtext.ide.server.build.XClusteringStorageAwareResourceLoader.executeClustered(XClusteringStorageAwareResourceLoader.java:107)
	at org.eclipse.n4js.xtext.ide.server.build.XBuildContext.executeClustered(XBuildContext.java:62)
	at org.eclipse.n4js.xtext.ide.server.build.XStatefulIncrementalBuilder.launch(XStatefulIncrementalBuilder.java:154)
	at org.eclipse.n4js.xtext.ide.server.build.XIncrementalBuilder.build(XIncrementalBuilder.java:60)
	at org.eclipse.n4js.xtext.ide.server.build.XIncrementalBuilder.build(XIncrementalBuilder.java:42)
	at org.eclipse.n4js.xtext.ide.server.build.ProjectBuilder.doBuild(ProjectBuilder.java:360)
	at org.eclipse.n4js.ide.server.build.N4JSProjectBuilder.doBuild(N4JSProjectBuilder.java:81)
	at org.eclipse.n4js.xtext.ide.server.build.ProjectBuilder.doIncrementalBuild(ProjectBuilder.java:327)
	at org.eclipse.n4js.xtext.ide.server.build.XWorkspaceBuilder.doIncrementalBuild(XWorkspaceBuilder.java:536)
	at org.eclipse.n4js.xtext.ide.server.build.XWorkspaceBuilder.doIncrementalWorkspaceUpdateAndBuild(XWorkspaceBuilder.java:456)
	at org.eclipse.n4js.xtext.ide.server.build.BuilderFrontend.lambda$6(BuilderFrontend.java:191)
	at org.eclipse.n4js.ide.server.concurrent.N4JSQueuedExecutorService.lambda$0(N4JSQueuedExecutorService.java:35)
	at org.eclipse.n4js.xtext.ide.server.QueuedExecutorService$QueuedTask.run(QueuedExecutorService.java:124)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: element to process may not be null
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1241)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.processBinaryExpression(PrettyPrinterEcmaScript.java:1404)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseBinaryLogicalExpression(PrettyPrinterEcmaScript.java:1083)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseBinaryLogicalExpression(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:1378)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1243)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.processBinaryExpression(PrettyPrinterEcmaScript.java:1400)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseBinaryLogicalExpression(PrettyPrinterEcmaScript.java:1083)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseBinaryLogicalExpression(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:1378)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1243)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseVariableDeclaration(PrettyPrinterEcmaScript.java:440)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseVariableDeclaration(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:480)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1216)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseVariableStatement(PrettyPrinterEcmaScript.java:408)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseVariableStatement(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:450)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1243)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.process(PrettyPrinterEcmaScript.java:1226)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseScript(PrettyPrinterEcmaScript.java:101)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.caseScript(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.n4js.n4JS.util.N4JSSwitch.doSwitch(N4JSSwitch.java:96)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:82)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:58)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:75)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.doSwitch(PrettyPrinterEcmaScript.java:1)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.n4js.transpiler.print.PrettyPrinterEcmaScript.append(PrettyPrinterEcmaScript.java:55)
	at org.eclipse.n4js.transpiler.print.PrettyPrinter.print(PrettyPrinter.java:60)
	at org.eclipse.n4js.transpiler.PrettyPrintingStep.print(PrettyPrintingStep.java:37)
	at org.eclipse.n4js.transpiler.AbstractTranspiler.prettyPrint(AbstractTranspiler.java:273)
	at org.eclipse.n4js.transpiler.AbstractTranspiler.transpile(AbstractTranspiler.java:169)
	at org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator.internalDoGenerate(EcmaScriptSubGenerator.java:163)
	at org.eclipse.n4js.generator.AbstractSubGenerator.doGenerate(AbstractSubGenerator.java:153)
	... 29 more

```

--- a/plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/ASTStructureValidator.xtend
+++ b/plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/ASTStructureValidator.xtend
@@ -1836,6 +1836,12 @@ class ASTStructureValidator {
 					IssueCodes.AST_BINARY_LOGICAL_EXPRESSION_MISSING_PART))
 
 		}
+		recursiveValidateASTStructure(
+			model,
+			producer,
+			validLabels,
+			constraints
+		);
 	}
 
 	def private dispatch void validateASTStructure(

--- a/tests/org.eclipse.n4js.transpiler.es5.tests/testdata/expressions/BinaryExpression.n4js.xt
+++ b/tests/org.eclipse.n4js.transpiler.es5.tests/testdata/expressions/BinaryExpression.n4js.xt
@@ -1,0 +1,19 @@
+/*
+ * Copyright (c) 2016 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+
+/* XPECT_SETUP org.eclipse.n4js.transpiler.es5.xpect.tests.CompileAndExecuteTest END_SETUP */
+
+
+// XPECT errors --> "The logical expression is missing the right operand." at "true ||"
+const b = true ||
+    || false
+;
+console.log(b);

